rq/queue.py

```diff
@@ -458,8 +457,19 @@ def enqueue_dependents(self, job, pipeline=None):
                 if pipeline is None:
                     pipe.watch(dependents_key)
 
-                dependent_jobs = [self.job_class.fetch(as_text(job_id), connection=self.connection)
-                                  for job_id in pipe.smembers(dependents_key)]
+                dependent_job_ids = [as_text(_id)
+                                     for _id in pipe.smembers(dependents_key)]
+
+                dependent_jobs = [
+                    dependent_job for dependent_job
+                    in self.job_class.fetch_many(
+                        dependent_job_ids,
+                        connection=self.connection
+                    ) if dependent_job.dependencies_are_met(
+                        pipeline=pipe,
+                        exclude={job.id}
```

Comment:
Does exclude need to be a set? I think it can be a single `job_id`

Reply:
:heavy_check_mark: in 24d2a399164adc6214df71f1507b84d05b8af789